### PR TITLE
Fix Help Text Parse Order

### PR DESCRIPTION
**Pull request by siddk**
Adds ability to set a "preferred" help text parse order from call to `draccus.wrap(preferred_help="above")` or `draccus.parse(..., preferred_help="inline")`.

Additionally changes the default help text parse order to "inline" --> "above" --> "below" (vs. the current "below"-first default).

### CODE REVIEW

File: draccus/argparsing.py [hunk 1]
```diff
@@ -78,9 +82,16 @@ def _set_dataclass(
             default = dataclass if default is None else default
             dataclass = type(dataclass)
 
-        new_wrapper = dataclass_wrapper_class(dataclass, default=default)
+        new_wrapper = dataclass_wrapper_class(dataclass, default=default, preferred_help=self.preferred_help)
         new_wrapper.register_actions(parser=self.parser)
 
+    def _assert_preferred_help(self):
+        """Checks that `self.prefer_help` is valid."""
+        if self.preferred_help not in {"inline", "above", "below"}:
```
Comment (dlwh) [inline draccus/argparsing.py:90]: does it make sense to break out a stringenum?

File: draccus/wrappers/docstring.py [hunk 2]
```diff
@@ -18,6 +18,28 @@ class AttributeDocString:
...
+# Default Help Orderings
+HELP_ORDERS = {
+    "inline": ["comment_inline", "comment_above", "docstring_below"],
+    "above": ["comment_above", "comment_inline", "docstring_below"],
+    "below": ["docstring_below", "comment_inline", "comment_above"],
+}
+
+
+def get_preferred_help_text(doc: AttributeDocString, preferred_help: str = "inline") -> Optional[str]:
+    if doc is None:
```
Comment (dlwh) [inline draccus/wrappers/docstring.py:30]: should it be Optional[AttributeDocString] then?

File: pyproject.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "draccus"
-version = "0.6"
+version = "0.1"
```
Comment (dlwh) [inline pyproject.toml:7]: revert

File: draccus/wrappers/docstring.py [hunk 4]
```diff
@@ -18,6 +18,28 @@ class AttributeDocString:
     docstring_below: str = ""
 
 
+# Default Help Orderings
+HELP_ORDERS = {
+    "inline": ["comment_inline", "comment_above", "docstring_below"],
```
Comment (dlwh) [inline draccus/wrappers/docstring.py:23]: imho docstring_below should be the first backup for this and "above": if you're putting a string right after a field, it is probably intended as the doc for the field

File: draccus/argparsing.py (hunk 1 above)
Comment (siddk) [reply draccus/argparsing.py:90]: Feels like more overhead? Then if you're overloading `draccus.wraps()` it's an extra import (`from draccus import Help; draccus.wrap(Help.INLINE)`? Your call though

File: draccus/wrappers/docstring.py (hunk 2 above)
Comment (siddk) [reply draccus/wrappers/docstring.py:30]: Oh good catch; by the semantics of `get_attribute_docstring` you'll never get a null AttributeDocString, it'll just be empty. Updating.

File: draccus/wrappers/docstring.py (hunk 4 above)
Comment (siddk) [reply draccus/wrappers/docstring.py]: 10-4, resolved.

File: draccus/argparsing.py (hunk 1 above)
Comment (dlwh) [reply draccus/argparsing.py:90]: i was thinking stringenum so you could choose between the two (HF does things this way frequently)

Comment (siddk) [reply draccus/argparsing.py:91]: Fixed... I think?

Comment (dlwh) [reply draccus/argparsing.py:91]: i meant something much simpler really. I went in and did it. https://github.com/dlwh/draccus/pull/9/commits/1f6ca95387b6505f3d1e5697125a4c0e2c6159af#diff-a524e53291b158cc810ec980f25dfcf430909d75a6e1fc0de59b2cfe4a1656b8

**Review by dlwh — APPROVED** [review]

File: draccus/argparsing.py (hunk 1 above)
Comment (siddk) [reply draccus/argparsing.py:91]: Oh I totally missed this - thanks David!

**Pull request merged**